packages/babel-plugin-transform-react-pure-annotations/src/index.ts

```diff
@@ -60,19 +61,17 @@ function isReactCall(path) {
   // a default import (`import React from 'react'`) or namespace
   // import (`import * as React from 'react'), and check if the
   // property matches one of the pure methods.
-  for (const [module, methods] of PURE_CALLS) {
-    const object = path.get("callee.object");
-    if (
-      object.referencesImport(module, "default") ||
-      object.referencesImport(module, "*")
-    ) {
-      for (const method of methods) {
-        if (t.isIdentifier(path.node.callee.property, { name: method })) {
-          return true;
-        }
+  const object = calleePath.get("object");
+  const callee = calleePath.node;
+  if (!callee.computed && t.isIdentifier(callee.property)) {
```

Comment:
Note: this computed check is the only real fix, everything else is a minor refactor.